test/bpf/Makefile

```diff
@@ -30,6 +32,15 @@ elf-demo.o: elf-demo.c
 	@$(ECHO_CC)
 	$(CLANG) ${FLAGS_CLANG} ${FLAGS} -I../../bpf/ $< -o $@
 
+nat-test: nat-test.c $(LIB)
+	@$(ECHO_CC)
+	make -C $(ROOT_DIR)/bpf/mock mock_customized filename=conntrack_stub.h
+	$(CLANG) ${FLAGS} -I $(ROOT_DIR)/bpf/ -Icmock  $< cmock/unity.c cmock/cmock.c $(ROOT_DIR)/bpf/mock/mock_helpers.c $(ROOT_DIR)/bpf/mock/mock_helpers_skb.c $(ROOT_DIR)/bpf/mock/mock_conntrack_stub.c -o $@
+
+unit-tests: $(ALL_TESTS)
+	@$(ECHO_CHECK)
+	$(foreach test,$<,$(ROOT_DIR)/$(RELATIVE_DIR)/$(test))
```

Comment:
Hmm, I thought this should work, but when I try it now that nat-test is in the $ALL_TESTS, it doesn't seem like the nat test is getting run. The iteration doesn't work the way I expected (I expected `$<` to expand to `unit-test nat-test` and then this iteration would run each of the binaries).

```
$ remake -x -C test/bpf unit-tests
Reading makefiles...
remake: Entering directory '/home/joe/git/cilium-review/test/bpf'
Updating makefiles...
Updating goal targets...
 File 'unit-tests' does not exist.
Must remake target 'unit-tests'.
Makefile:41: update target 'unit-tests' due to: unit-test nat-test
##>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
echo "  CHECK test/bpf"
##<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
  CHECK test/bpf
##>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
/home/joe/git/cilium-review/test/bpf/unit-test
##<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Successfully remade target file 'unit-tests'.
remake: Leaving directory '/home/joe/git/cilium-review/test/bpf'
```

Reply:
^^ OK, I suggested the wrong variable. `$<` is the first dependency, we actually need `$^`:

https://www.gnu.org/software/make/manual/html_node/Automatic-Variables.html

Reply:
Thanks! I just tried this and it worked. Updated!